2010 South Carolina Congressional Districts

Jfer09 opened the pull request:
## Redistricting requirements
In South Carolina, districts must:

1. be contiguous
1. have equal populations
1. be geographically compact
1. preserve county and municipality boundaries as much as possible


### Interpretation of requirements
We enforce a maximum population deviation of 0.5%.

## Data Sources
Data for South Carolina comes from the ALARM Project's [2020 Redistricting Data Files](https://alarm-redist.github.io/posts/2021-08-10-census-2020/).

## Pre-processing Notes
No manual pre-processing decisions were necessary.

## Simulation Notes
We sample 6,000 districting plans for South Carolina.
We impose a hinge constraint on the Black Voting Age Population so that it encourages district BVAP between 40 percent and 60 percent, but districts with BVAP of 30% or less are not penalized as much. 
We impose a municipality-split constraint to lower the number of municipality splits.

## Validation

![this one](https://user-images.githubusercontent.com/66656384/175795407-6d7a059e-fb95-41fb-9254-906e839f9451.png)

<img width="1440" alt="Screen Shot 2022-06-25 at 9 21 39 PM" src="https://user-images.githubusercontent.com/66656384/175795467-3a5d7387-4dff-47ca-86ba-1798df4291e4.png">

```
R-hat values for summary statistics:
   pop_overlap      total_vap       plan_dev      comp_edge    comp_polsby       pop_hisp 
     1.0040012      1.0105256      1.0004111      1.0140118      1.0031822      1.0199507 
     pop_white      pop_black       pop_aian      pop_asian       pop_nhpi      pop_other 
     1.0037479      1.0033236      1.0095031      1.0056256      1.0028691      0.9999751 
       pop_two       vap_hisp      vap_white      vap_black       vap_aian      vap_asian 
     1.0054153      1.0131019      1.0003802      1.0036146      1.0115376      1.0064758 
      vap_nhpi      vap_other        vap_two pre_16_dem_cli pre_16_rep_tru uss_16_dem_dix 
     1.0011768      1.0016542      1.0010121      1.0054428      1.0061057      1.0029835 
uss_16_rep_sco gov_18_dem_smi gov_18_rep_mcm atg_18_dem_ana atg_18_rep_wil sos_18_dem_whi 
     0.9998713      1.0108486      1.0058166      1.0099854      1.0042444      1.0137063 
sos_18_rep_ham pre_20_rep_tru pre_20_dem_bid uss_20_rep_gra uss_20_dem_har         arv_16 
     1.0053737      1.0016564      1.0046326      1.0005294      1.0049484      1.0012903 
        adv_16         arv_18         adv_18         arv_20         adv_20  county_splits 
     1.0050325      1.0051256      1.0124208      1.0009286      1.0049237      1.0160643 
   muni_splits            ndv            nrv        ndshare        e_dvs.x       pr_dem.x 
     1.0115901      1.0099238      1.0030191      1.0002831      1.0002762      1.0103265 
       e_dem.x        pbias.x         egap.x        e_dvs.y       pr_dem.y        e_dem.y 
     1.0023588      1.0111069      1.0025052      1.0002762      1.0103265      1.0023588 
       pbias.y         egap.y 
     1.0111069      1.0025052 

Sampling diagnostics for SMC run 1 of 2 (3,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     2,906 (96.9%)     12.8%        0.27 1,900 (100%)      8 
Split 2     2,787 (92.9%)     18.6%        0.41 1,840 ( 97%)      5 
Split 3     2,671 (89.0%)     20.2%        0.47 1,854 ( 98%)      4 
Split 4     2,502 (83.4%)     23.1%        0.55 1,768 ( 93%)      3 
Split 5     2,408 (80.3%)     25.5%        0.63 1,710 ( 90%)      2 
Split 6     2,283 (76.1%)      9.1%        0.65 1,512 ( 80%)      2 
Resample    1,139 (38.0%)       NA%        1.13 1,447 ( 76%)     NA 

Sampling diagnostics for SMC run 2 of 2 (3,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     2,905 (96.8%)     11.3%        0.27 1,916 (101%)      9 
Split 2     2,783 (92.8%)     15.5%        0.39 1,864 ( 98%)      6 
Split 3     2,661 (88.7%)     21.4%        0.47 1,844 ( 97%)      4 
Split 4     2,535 (84.5%)     23.6%        0.57 1,807 ( 95%)      3 
Split 5     2,404 (80.1%)     25.3%        0.63 1,690 ( 89%)      2 
Split 6     2,304 (76.8%)      8.4%        0.63 1,476 ( 78%)      2 
Resample    1,245 (41.5%)       NA%        1.14 1,464 ( 77%)     NA 
```

## Checklist

- [x] I have followed the [instructions](https://github.com/alarm-redist/fifty-states/blob/main/CONTRIBUTING.md)
- [x] I have updated the [tracker](https://docs.google.com/spreadsheets/d/1k_tYLoE49W_DCK1tcWbouoYZFI9WD76oayEt5TOmJg4/edit#gid=453387933)
- [x] All `TODO` lines from the template code have been removed
- [x] I have merged in the master branch and then recalculated summary statistics
- [x] I have run `enforce_style()` to format my code
- [x] The documentation copied above is up-to-date 
- [x] There are no data files in this pull request
- [x] None of the file output paths (for the `redist_map` and `redist_plans` objects, and summary statistics) have been edited

@CoryMcCartan
@christopherkenny

kuriwaki commented on analyses/SC_cd_2010/01_prep_SC_cd_2010.R:
```diff
@@ -0,0 +1,91 @@
+# Download necessary files for analysis -----
+cli_process_start("Downloading files for {.pkg SC_cd_2010}")
+
+path_data <- download_redistricting_file("SC", "data-raw/SC")
```
This should have an argument, `year = 2010`. Otherwise the default is `year = 2020` and it is currently downloading 2020 Census statistics I believe

kuriwaki commented on analyses/SC_cd_2010/01_prep_SC_cd_2010.R:
```diff
@@ -0,0 +1,91 @@
+    cli_process_start("Preparing {.strong SC} shapefile")
+    # read in redistricting data
+    sc_shp <- read_csv(here(path_data), col_types = cols(GEOID20 = "c")) %>%
+        join_vtd_shapefile() %>%
```
add `year = 2010` to this function too

kuriwaki commented on analyses/SC_cd_2010/01_prep_SC_cd_2010.R:
```diff
@@ -0,0 +1,91 @@
+    # add the enacted plan
+    cd_shp <- st_read(here(path_enacted))
+    sc_shp <- sc_shp %>%
+        mutate(cd_2020 = as.integer(cd_shp$DISTRICT)[
```
It seems like `path_enacted` is the map enacted in 2010 (for the 2012 elections)? Then I think it should be renamed cd_2010. We should check if it is identical to `cd_2010` that already comes with data and disambiguate.

kuriwaki replied:
Also, I think this should be `cd_shp$District` instead of `$DISTRICT` -- that's what the column names look like when I read in `cd_shp` in the line above.

kuriwaki commented on analyses/SC_cd_2010/doc_SC_cd_2010.md:
```diff
@@ -0,0 +1,23 @@
+# 2010 South Carolina Congressional Districts
+
+## Redistricting requirements
+In South Carolina, districts must:
```
Can you add a link to any documents here? If it is old documents from links that are about to decay, you could potentially use waybackmachine

kuriwaki commented on analyses/SC_cd_2010/doc_SC_cd_2010.md:
```diff
@@ -0,0 +1,23 @@
+1. have equal populations
+1. be geographically compact
+1. preserve county and municipality boundaries as much as possible
+
```
I would also note that 2010 was before the Shelby County court ruling so South Carolina's plans had to pass pre-clearance from the DOJ.  I don't think we'd change the simulation per se, but good to note.

kuriwaki commented on analyses/SC_cd_2010/doc_SC_cd_2010.md:
```diff
@@ -0,0 +1,23 @@
+
+
+### Interpretation of requirements
+We enforce a maximum population deviation of X.X%.
```
Fill in when final

kuriwaki commented on analyses/SC_cd_2010/03_sim_SC_cd_2010.R:
```diff
@@ -0,0 +1,63 @@
+    pop_temper =  0.01,
+    counties = county,
+    constraints = constr_sc) %>%
+    match_numbers("cd_2010")
```
Once the `summary(plans)` output looks good, we will be thinning this 6000 plans to 5000. You can copy from other 2020 code that has been merged in.

kuriwaki commented on analyses/SC_cd_2010/doc_SC_cd_2010.md:
```diff
@@ -0,0 +1,23 @@
+No manual pre-processing decisions were necessary.
+
+## Simulation Notes
+We sample 5,000 districting plans for South Carolina.
```
This will be the value of `nsims`* `runs` in the simulation. So in this case, 6000.

kuriwaki reviewed (changes requested):
Thanks @Jfer09, looking good. Looks like the Census data that you loaded is from 2020 (the silent default), not 2010. I've marked places that should have an explicit `year = 2010` argument. You may need to delete some intermediate files to make sure you've replaced them with 2010 data.  

A sanity check of whether these are 2010 data is checking the population deviation in the validation plot. cd_2010 should have a population deviation of 1% of or less, but your histogram has a red line at nearly 10%. That suggests 2010 lines are evaluated on  population data that is from another decade.

kuriwaki commented on analyses/SC_cd_2010/01_prep_SC_cd_2010.R:
```diff
@@ -0,0 +1,91 @@
+        select(-vtd)
+    d_cd <- make_from_baf("SC", "CD", "VTD")  %>%
+        transmute(GEOID = paste0(censable::match_fips("SC"), vtd),
+            cd_2010 = as.integer(cd))
```
I think this should be called `cd_2000` as in `cd_2000 = as.integer(cd)`. A bit confusing, but the year after cd should be the year corresponding to the cycle. So the lines that was enacted in 2011 for the 2012 elections is called cd_2010. The lines that come from `make_from_baf` are lines before the redistricting, so they are 2000 cycle lines

Jfer09 commented:
# 2010 South Carolina Congressional Districts

## Redistricting requirements
In South Carolina, districts must:

1. be contiguous
2. have equal populations
3. be geographically compact
4. preserve county and municipality boundaries as much as possible
5. pass pre-clearance from the DOJ

https://redistricting.scsenate.gov/Documents/RedistrictingGuidelinesAdopted041311.pdf
https://redistricting.schouse.gov/archives/2011/6334-1500-2011-Redistricting-Guidelines-(A0404871).pdf

### Interpretation of requirements
We do not adhere to all criteria in the guidelines. We include the following constraints:

1. We enforce a maximum population deviation of 0.5%.
2. We impose a hinge constraint on the Black Voting Age Population so that it encourages districts with BVAP above 40%, but districts with BVAP of 30% or less are not penalized as much.
3. We impose a municipality-split constraint to lower the number of municipality splits.

## Data Sources
Data for South Carolina comes from the ALARM Project's [2020 Redistricting Data Files](https://alarm-redist.github.io/posts/2021-08-10-census-2020/). <- not sure what I should put for 2010 because I couldn't find it in the ALARM website :(

## Pre-processing Notes
No manual pre-processing decisions were necessary.

## Simulation Notes
We sample 6,000 districting plans for South Carolina.
No special techniques were needed to produce the sample.

## Validation

![validation_20220706_2203](https://user-images.githubusercontent.com/66656384/177677705-d99e8688-01f4-4425-8e32-bb95bcd88099.png)

![partisanship_of_seats](https://user-images.githubusercontent.com/66656384/177678082-3880bff9-b638-4c17-8c15-26a4d666bd54.png)


```
R-hat values for summary statistics:
   pop_overlap      total_vap       plan_dev      comp_edge    comp_polsby      pop_white      pop_black       pop_hisp 
     1.0017184      1.0076736      1.0000837      1.0014655      1.0080891      1.0053784      1.0150619      1.0100801 
      pop_aian      pop_asian       pop_nhpi      pop_other        pop_two      vap_white      vap_black       vap_hisp 
     0.9998978      1.0005043      1.0024173      1.0025930      1.0016946      1.0043995      1.0126980      1.0093582 
      vap_aian      vap_asian       vap_nhpi      vap_other        vap_two pre_16_rep_tru pre_16_dem_cli pre_20_rep_tru 
     1.0001556      1.0006002      1.0041020      1.0017029      1.0025194      1.0040227      1.0061493      1.0035632 
pre_20_dem_bid uss_16_rep_sco uss_16_dem_dix uss_20_rep_gra uss_20_dem_har gov_18_dem_smi gov_18_rep_mcm atg_18_dem_ana 
     1.0013353      1.0022584      1.0117500      1.0052989      1.0006937      1.0032953      1.0096629      1.0029299 
atg_18_rep_wil sos_18_dem_whi sos_18_rep_ham         adv_16         adv_18         adv_20         arv_16         arv_18 
     1.0085896      1.0078122      1.0090309      1.0093736      1.0033046      1.0016536      1.0023676      1.0089577 
        arv_20  county_splits    muni_splits            ndv            nrv        ndshare          e_dvs         pr_dem 
     1.0043677      1.0082489      1.0058738      1.0012658      1.0053194      1.0049363      1.0049391      1.0009597 
         e_dem          pbias           egap 
     1.0021474      1.0082487      1.0003794 

Sampling diagnostics for SMC run 1 of 2 (3,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     2,850 (95.0%)     11.1%        0.32 1,882 ( 99%)      9 
Split 2     2,722 (90.7%)     15.1%        0.44 1,839 ( 97%)      6 
Split 3     2,637 (87.9%)     20.0%        0.50 1,792 ( 94%)      4 
Split 4     2,554 (85.1%)     14.7%        0.57 1,779 ( 94%)      5 
Split 5     2,398 (79.9%)     18.4%        0.63 1,758 ( 93%)      3 
Split 6     2,257 (75.2%)      8.2%        0.69 1,475 ( 78%)      2 
Resample    1,186 (39.5%)       NA%        1.22 1,454 ( 77%)     NA 

Sampling diagnostics for SMC run 2 of 2 (3,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     2,849 (95.0%)     14.4%        0.33 1,902 (100%)      7 
Split 2     2,719 (90.6%)     18.5%        0.43 1,854 ( 98%)      5 
Split 3     2,607 (86.9%)     20.3%        0.53 1,825 ( 96%)      4 
Split 4     2,525 (84.2%)     23.0%        0.62 1,764 ( 93%)      3 
Split 5     2,418 (80.6%)     24.3%        0.64 1,767 ( 93%)      2 
Split 6     2,282 (76.1%)      8.2%        0.63 1,521 ( 80%)      2 
Resample    1,241 (41.4%)       NA%        1.16 1,453 ( 77%)     NA 
```

## Checklist

- [x] I have followed the [instructions](https://github.com/alarm-redist/fifty-states/blob/main/CONTRIBUTING.md)
- [x] I have updated the [tracker](https://docs.google.com/spreadsheets/d/1k_tYLoE49W_DCK1tcWbouoYZFI9WD76oayEt5TOmJg4/edit#gid=453387933)
- [x] All `TODO` lines from the template code have been removed
- [x] I have merged in the master branch and then recalculated summary statistics
- [x] I have run `enforce_style()` to format my code
- [x] The documentation copied above is up-to-date 
- [x] There are no data files in this pull request
- [x] None of the file output paths (for the `redist_map` and `redist_plans` objects, and summary statistics) have been edited

@kuriwaki

kuriwaki commented:
The performance plot is giving signs that the most Black district ("Ordered district 7") in a decent chunk of the plans are not performing, i.e. they are more Republican than Democrat. That might not be compliant, especially given that 2010 districts would have fallen under pre-clearance. So we may need to  strengthen the constraints so we are left with all performant districts.

@Jfer09 can we try summary statistics like the below and see what the numbers are? You can check out how it was summarized in 2020 in the [PR post](https://github.com/alarm-redist/fifty-states/pull/85#issue-1256384530), right under the performance plot. 

https://github.com/alarm-redist/fifty-states/blob/5165d8aa073fa6a6fc07aa0f2fca014972b2a6e1/analyses/SC_cd_2020/03_sim_SC_cd_2020.R#L79-L94

Jfer09 commented:
# 2010 South Carolina Congressional Districts

## Redistricting requirements
In South Carolina, districts must:

1. be contiguous
2. have equal populations
3. be geographically compact
4. preserve county and municipality boundaries as much as possible
5. pass pre-clearance from the DOJ

https://redistricting.scsenate.gov/Documents/RedistrictingGuidelinesAdopted041311.pdf
https://redistricting.schouse.gov/archives/2011/6334-1500-2011-Redistricting-Guidelines-(A0404871).pdf

### Interpretation of requirements
We do not adhere to all criteria in the guidelines. We include the following constraints:

1. We enforce a maximum population deviation of 0.5%.
2. We impose a hinge constraint on the Black Voting Age Population so that it encourages districts with BVAP above 50%, but districts with BVAP of 30% or less are not penalized as much. This ensures that districts with high BVAP are able to elect their candidate of choice. 
3. We impose a municipality-split constraint to lower the number of municipality splits.

## Data Sources
Data for South Carolina comes from the ALARM Project's [2020 Redistricting Data Files](https://alarm-redist.github.io/posts/2021-08-10-census-2020/). <- not sure what I should put for 2010 because I couldn't find it in the ALARM website :(

## Pre-processing Notes
No manual pre-processing decisions were necessary.

## Simulation Notes
We sample 6,000 districting plans across two independent runs of the SMC algorithm. We then remove all plans that do not contain any district that has both a BVAP of over 30% and an average vote share that is more Democratic than Republican. This removal occurs after verifying that such plans comprise less than 1% of the 6,000 plans. We then thin the sample down to exactly 5,000 plans. We also set the population tempering to 0.01 to avoid bottlenecks.

## Validation
![validation_20220804_1452](https://user-images.githubusercontent.com/66656384/182931203-486948da-6185-4034-ae8d-7d615a69b944.png)
![Rplot](https://user-images.githubusercontent.com/66656384/182931159-5c25901d-45e7-4891-8b26-4811700a2b47.jpeg)

91% of the highest-BVAP district and 28.5% of the second highest-BVAP district are blue in this plot.

``` 
n_black_perf    n
1            1 3996
2            2  997
3            3    7
```
Summary of Plan Object:

```
SMC: 6,000 sampled plans of 7 districts on 2,122 units
`adapt_k_thresh`=0.985 • `seq_alpha`=0.5
`est_label_mult`=1 • `pop_temper`=0.01

Plan diversity 80% range: 0.48 to 0.83

R-hat values for summary statistics:
   pop_overlap      total_vap       plan_dev      comp_edge    comp_polsby      pop_white 
     1.0120475      1.0042467      1.0016465      1.0014947      1.0304959      1.0015804 
     pop_black       pop_hisp       pop_aian      pop_asian       pop_nhpi      pop_other 
     1.0050795      1.0005244      1.0076705      1.0002316      1.0035516      1.0016537 
       pop_two      vap_white      vap_black       vap_hisp       vap_aian      vap_asian 
     1.0008869      1.0011788      1.0051134      1.0049364      1.0080004      1.0002082 
      vap_nhpi      vap_other        vap_two pre_16_rep_tru pre_16_dem_cli pre_20_rep_tru 
     1.0034503      1.0010427      0.9999715      1.0154680      1.0074157      1.0092683 
pre_20_dem_bid uss_16_rep_sco uss_16_dem_dix uss_20_rep_gra uss_20_dem_har gov_18_dem_smi 
     1.0037990      1.0051767      1.0055020      1.0078760      1.0057119      1.0081538 
gov_18_rep_mcm atg_18_dem_ana atg_18_rep_wil sos_18_dem_whi sos_18_rep_ham         adv_16 
     1.0137191      1.0108854      1.0127145      1.0109889      1.0141728      1.0069088 
        adv_18         adv_20         arv_16         arv_18         arv_20  county_splits 
     1.0104412      1.0051354      1.0107766      1.0138879      1.0088795      1.0093744 
   muni_splits            ndv            nrv        ndshare          e_dvs         pr_dem 
     1.0178542      1.0098929      1.0118179      1.0123676      1.0121183      1.0020453 
         e_dem          pbias           egap 
     1.0109351      1.0076293      1.0278558 

Sampling diagnostics for SMC run 1 of 2 (3,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,569 (52.3%)     11.1%        0.32 1,882 ( 99%)      9 
Split 2     1,998 (66.6%)     16.2%        0.90 1,490 ( 79%)      6 
Split 3     2,064 (68.8%)     20.5%        0.70 1,636 ( 86%)      4 
Split 4     2,094 (69.8%)     23.6%        0.70 1,726 ( 91%)      3 
Split 5     1,953 (65.1%)     24.3%        0.71 1,679 ( 89%)      2 
Split 6     1,678 (55.9%)      8.4%        0.81 1,436 ( 76%)      2 
Resample      358 (11.9%)       NA%        1.54 1,136 ( 60%)     NA 

Sampling diagnostics for SMC run 2 of 2 (3,000 samples)
         Eff. samples (%) Acc. rate Log wgt. sd  Max. unique Est. k 
Split 1     1,595 (53.2%)     14.4%        0.33 1,902 (100%)      7 
Split 2     1,971 (65.7%)     20.0%        0.88 1,513 ( 80%)      5 
Split 3     2,024 (67.5%)     21.2%        0.72 1,631 ( 86%)      4 
Split 4     2,093 (69.8%)     23.3%        0.71 1,681 ( 89%)      3 
Split 5     2,107 (70.2%)     18.2%        0.71 1,681 ( 89%)      3 
Split 6     1,654 (55.1%)      8.8%        0.74 1,473 ( 78%)      2 
Resample       235 (7.8%)       NA%        1.48 1,065 ( 56%)     NA 
```


## Checklist

- [x] I have followed the [instructions](https://github.com/alarm-redist/fifty-states/blob/main/CONTRIBUTING.md)
- [x] I have updated the [tracker](https://docs.google.com/spreadsheets/d/1k_tYLoE49W_DCK1tcWbouoYZFI9WD76oayEt5TOmJg4/edit#gid=453387933)
- [x] All `TODO` lines from the template code have been removed
- [x] I have merged in the master branch and then recalculated summary statistics
- [x] I have run `enforce_style()` to format my code
- [x] The documentation copied above is up-to-date 
- [x] There are no data files in this pull request
- [x] None of the file output paths (for the `redist_map` and `redist_plans` objects, and summary statistics) have been edited

@kuriwaki
